src/types/fixedtimezone.jl

```diff
@@ -30,7 +34,7 @@ const FIXED_TIME_ZONE_REGEX = r"""
 A `TimeZone` with a constant offset for all of time.
 """
 struct FixedTimeZone <: TimeZone
-    name::String
+    name::FixedTimeZoneName
```

Comment:
Once ShortStrings.jl this type alias won't survive. May as well try to indicate what it truly is.
```suggestion
    name::_ShortString15
```